Cabal/Distribution/Compat/CopyFile.hs

```diff
@@ -25,20 +27,36 @@ import System.IO.Error
 import System.Directory
          ( doesFileExist, renameFile, removeFile )
 import System.FilePath
-         ( takeDirectory )
```

Comment:
This breaks the build on non-Windows, where `takeDirectory` is still used, see e.g. https://travis-ci.org/haskell/cabal/jobs/321706530#L778.